### PR TITLE
Make sync Job test tables more readable

### PR DESCRIPTION
**Pull request by alculquicondor**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Adds keys and deletes zero values from test table structs. This improves readability and makes it easier to add more parameters.

Also makes use of `t.Run` to improve debugging experience.

Added test for failed pod recreation (in separate commit).

**Notes to reviewer**:

Change was mostly performed by IDE.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/sig apps

### CODE REVIEW

File: pkg/controller/job/job_controller_test.go [hunk 1]
```diff
@@ -163,206 +163,323 @@ func TestControllerSyncJob(t *testing.T) {
...
+			// We need requeue syncJob task if podController error
+			if tc.podControllerError != nil {
+				if err == nil {
+					t.Error("Syncing jobs expected to return error on podControl exception")
+				}
+			} else if tc.failedPods > 0 && tc.expectedCondition == nil {
+				if err == nil {
+					t.Error("Syncing jobs expected to return error when there are new failed pods and Job didn't finish")
+				}
+			} else if tc.podLimit != 0 && fakePodControl.CreateCallCount > tc.podLimit {
```
Comment (adtac) [inline pkg/controller/job/job_controller_test.go:418]:
```suggestion
			} else if tc.podLimit != 0 && fakePodControl.CreateCallCount >= tc.podLimit {
```

right?

Comment (alculquicondor) [reply pkg/controller/job/job_controller_test.go:418]: no, it was rather a bug in the existing code :)

**Pull request merged**